### PR TITLE
add documentation for tidb_opt_distinct_agg_push_down

### PR DESCRIPTION
**Pull request by SunRunAway**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/2220
- Other reference link(s):https://github.com/pingcap/tidb/issues/14623

### CODE REVIEW

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 1]
```diff
@@ -52,6 +52,15 @@ set @@global.tidb_distsql_scan_concurrency = 10;
...
 当查询中聚合操作执行很慢时，可以尝试设置该变量为 1。
 
+### tidb_opt_distinct_agg_push_down
+
+作用域：SESSION
+
+默认值：0
+
+这个变量用来设置优化器是否执行带有 Distinct 的聚合函数（比如 `select count(distinct a) from t`）下推到 Coprocessor 的优化操作。
+当查询中带有 Distinct 的聚合操作执行很慢时，可以尝试设置该变量为 1。
```
Comment (zz-jason) [inline reference/configuration/tidb-server/tidb-specific-variables.md:62]: 需要举一个例子来说明：如何确认生效了（distinct 下推了）

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 2]
```diff
@@ -52,6 +52,15 @@ set @@global.tidb_distsql_scan_concurrency = 10;
 这个变量用来设置优化器是否执行聚合函数下推到 Join 之前的优化操作。
 当查询中聚合操作执行很慢时，可以尝试设置该变量为 1。
 
+### tidb_opt_distinct_agg_push_down
+
+作用域：SESSION
+
+默认值：0
+
+这个变量用来设置优化器是否执行带有 Distinct 的聚合函数（比如 `select count(distinct a) from t`）下推到 Coprocessor 的优化操作。
```
Comment (ilovesoup) [inline reference/configuration/tidb-server/tidb-specific-variables.md:61]: 需要大致解释下下推的转换。

Comment (SunRunAway) [reply reference/configuration/tidb-server/tidb-specific-variables.md:61]: Updated. PTAL.

File: reference/configuration/tidb-server/tidb-specific-variables.md (hunk 1 above)
Comment (SunRunAway) [reply reference/configuration/tidb-server/tidb-specific-variables.md:62]: Updated. PTAL.

**Comment by TomShawn** [comment]
@ilovesoup PTAL again

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 3]
```diff
@@ -52,6 +52,33 @@ set @@global.tidb_distsql_scan_concurrency = 10;
...
+当查询中带有 Distinct 的聚合操作执行很慢时，可以尝试设置该变量为 1。
+
+例如，在如下执行计划中，`distinct a` 被下推到了 Coprocessor， 在 `HashAgg_5` 里新增里一个 group by 列 `test.t.a`。
+
+```
+mysql> set session tidb_opt_distinct_agg_push_down = 1;
+Query OK, 0 rows affected (0.00 sec)
+
+mysql> desc select count(distinct a) from test.t;
```
Comment (zz-jason) [inline reference/configuration/tidb-server/tidb-specific-variables.md:70]: 给一个下推前的执行计划吧，方便用户直观的了解下推的效果

Comment (SunRunAway) [reply reference/configuration/tidb-server/tidb-specific-variables.md:80]: Updated, PTAL.

**Pull request merged**